queries/query_builders.go

```diff
@@ -327,6 +327,58 @@ func BuildUpsertQueryMSSQL(dia drivers.Dialect, tableName string, primary, updat
 	return buf.String()
 }
 
+// BuildUpsertQueryCockroachDB builds a SQL statement string using the upsertData provided.
+func BuildUpsertQueryCockroachDB(dia drivers.Dialect, tableName string, updateOnConflict bool, ret, update, conflict, whitelist []string) string {
```

Comment:
We need to come up with a mechanism to break this out. I think it should be simple, each driver should be able to create its own
upsert query. And we should just provide all the information to it that we can, and then it can make the decision what to do with it.

It involves figuring out a superset of all the required info then executing the driver's method.